packages/covector/src/run.ts

```diff
@@ -219,8 +223,154 @@ export function* covector({
 
     return pkgCommandsRan;
   } else if (command === "preview") {
-    console.log('WIP');
-    return 'placeholder';
+    yield raceTime({ t: config.timeout });
+
+    let versionTemplate;
+    
+    switch(previewVersion){
+      case "date":
+        versionTemplate = `${Date.now()}`;
+        break;
+      case "sha":
+        versionTemplate = `${context.after.substring(0, 7)}`;
+        break;
+      default:
+        throw new Error(`Preview version template you specified, "${previewVersion}", is invalid.`)
+    };
+
+    const versionChanges = changesConsideringParents({
+      assembledChanges,
+      config,
+    });
+
+    //@ts-ignore
+    const versionCommands: PkgVersion[] = yield mergeChangesToConfig({
+      assembledChanges: versionChanges,
+      config,
+      command: "version",
+      dryRun,
+      filterPackages,
+    });
+
+    let pkgCommandsRan: Covector = Object.keys(config.packages).reduce(
+      (
+        pkgs: {
+          [k: string]: {
+            precommand: string | false;
+            command: string | false;
+            postcommand: string | false;
+            applied: string | false;
+          };
+        },
+        pkg: string
+      ) => {
+        pkgs[pkg] = {
+          precommand: false,
+          command: false,
+          postcommand: false,
+          applied: false,
+        };
+        return pkgs;
+      },
+      {}
+    );
+
+    pkgCommandsRan = yield attemptCommands({
+      cwd,
+      commands: versionCommands,
+      commandPrefix: "pre",
+      command: "version",
+      pkgCommandsRan,
+      dryRun,
+    });
+
+    const applied = yield apply({
+      //@ts-ignore
+      commands: versionCommands,
+      config,
+      cwd,
+      bump: !dryRun,
+      previewTemplate: versionTemplate,
+    });
+
+    pkgCommandsRan = applied.reduce(
+      (
+        pkgs: {
+          [k: string]: {
+            precommand: boolean;
+            command: boolean;
+            postcommand: boolean;
+            applied: object;
+          };
+        },
+        result: { name: string }
+      ) => {
+        pkgs[result.name].applied = result;
+        return pkgs;
+      },
+      pkgCommandsRan
+    );
+
+    pkgCommandsRan = yield attemptCommands({
+      cwd,
+      commands: versionCommands,
+      commandPrefix: "post",
+      command: "version",
+      pkgCommandsRan,
+      dryRun,
+    });
+    
+    const publishCommands: PkgPublish[] = yield mergeIntoConfig({
+      assembledChanges,
+      config,
+      command: "publish",
+      cwd,
+      dryRun,
+      filterPackages,
+    });
+
+    if (publishCommands.length === 0) {
+      console.log(`No commands configured to run on publish.`);
```

Comment:
I was following [this](https://github.com/jbolda/covector/blob/main/packages/covector/src/run.ts#L240-L243) template that's already there for `publish`.

We could change it to something like `No ${ command } command specified in covector's configuration file".

Reply:
Oh wait, I see. Are we just running the publish commands effectively? Maybe that does make sense in that context.

Reply:
> Are we just running the publish commands effectively?

? Since `preview` needs to run both `version` and `publish`, I swapped out the command variable with the appropriate string values.

> Maybe that does make sense in that context.

So keep it as is or should I change it to what I suggested?

Reply:
I think let's keep it as is. 👍